### PR TITLE
Eliminate the `rt` and `vectored` features from the `esp-hal` package

### LINKED ISSUE #1371: It is not possible to build the `esp-hal` package without the `rt` and `vectored` features enabled
`crate::interrupt::bind_interrupt()` and `crate::interrupt::enable()` are both completely unavailable when the `vectored` feature is disabled, resulting in a number of build errors.

@MabezDev @bjoernQ what is the best course of action here?

### PR DESCRIPTION
**Pull request by jessebraham**
As discussed with the team earlier, since these two features cannot be disabled, and are generally useful, we will simply eliminate the features and this functionality will always be enabled.

We can revisit this in the future if there is a need for it.

Closes #1371.

### CODE REVIEW

**Comment by burrbull** [comment]
maybe enable them by default in pac crates?

**Comment by MabezDev** [comment]
> maybe enable them by default in pac crates?

I thought about that too, but I think its best leave it opt-in for PAC crates. For instance, if we want to use the PACs directly in a probe-rs flasher stub, we won't have a runtime there.

File: esp-hal/src/interrupt/riscv.rs [hunk 1]
```diff
@@ -888,9 +873,8 @@ mod clic {
 
     /// Set the priority level of an CPU interrupt
     ///
-    /// Great care must be taken when using the `vectored` feature (enabled by
-    /// default). Avoid changing the priority of interrupts 1 - 15 when
-    /// interrupt vectoring is enabled.
+    /// Great care must be taken when using this funciton; avoid changing the
```
Comment (MabezDev) [inline esp-hal/src/interrupt/riscv.rs:876]:
```suggestion
    /// Great care must be taken when using this function; avoid changing the
```

**Review by MabezDev — APPROVED** [review]
LGTM, one small typo I found but that's about it!

File: esp-hal/src/interrupt/riscv.rs (hunk 1 above)
Comment (jessebraham) [reply esp-hal/src/interrupt/riscv.rs:876]: Oops, good eye!

**Review by bjoernQ — COMMENTED** [review]
Thanks!

**Pull request merged**